sidechain/consensus/slots/src/lib.rs

```diff
@@ -177,7 +186,7 @@ pub trait SimpleSlotWorker<B: ParentchainBlock> {
 	/// this function from the outside at each slot.
 	fn on_slot(
 		&mut self,
-		slot_info: SlotInfo<B>,
+		slot_info: SlotInfo<ParentchainBlock>,
 		shard: ShardIdentifierFor<Self::Output>,
 	) -> Option<SlotResult<Self::Output>> {
 		let (_timestamp, slot) = (slot_info.timestamp, slot_info.slot);
```

Comment:
Would it be possible (with reasonable effort) to write a test that failed before your changes, but passes now?